docs/languages/TypeScript.md

```diff
@@ -45,6 +46,20 @@ This can be done by including the preset `TS_COMMON_PRESET` using the option `ma
 
 Check out this [example out for a live demonstration](../../examples/typescript-generate-marshalling).
 
+## Generate models with jsonbinpack support
+
+This functionality is for the library [jsonbinpack](https://github.com/sourcemeta/jsonbinpack).
+
+This preset can ONLY be used with AsyncAPI 2.x and JSON Schema draft 4 to 7 inputs.
+
+This functionality has two requirements:
+1. You MUST manually install the library `jsonbinpack`.
+2. You MUST also use the [Generate un/marshal functions for classes](#generate-unmarshal-functions-for-classes)
+
+This feature enables you to conver the models into a buffer that is highly space efficient, this is instead of sending pure JSON data across the wire.
```

Comment:
```suggestion
This feature allows you to convert models to a buffer, which is highly space-efficient, instead of sending pure JSON data over the wire.
```